scripts/add_component_skeleton.py

```diff
@@ -0,0 +1,191 @@
+    "functional tests."
+)
+
+gate = (
+    "Gate jobs are triggered by a comment on a pull request. The change is "
+    "merged automatically if all the tests pass. The gate job ensures that "
+    "changes are tested against the state of the target branch when the code "
+    "is merged. In comparison, check jobs test against the state of the "
+    "branch when the test is started."
+)
+
+release = (
+    "Release jobs are used to validate that a particular commit can be used "
+    "for a new release of the project. They tend to be the most expensive "
+    "tests in terms of time and resources."
+)
+
+periodic = (
+    "Periodic jobs are triggered by a configurable timer, by default they run "
+    "daily and use the latest commit to the branch being tested."
+)
+
+rel_notes = (
+    "The release notes generated by this step are used when releasing a new "
+    "version of the project. The releases should document the changes between "
+    "the next version and the latest released."
+)
+
+dep_update = (
+    "Dependency updates are checked periodically. A project should codify the "
+    "process for updating the working directory of the repository with any "
+    "changes to upstream dependencies. If changes are made, the job will "
+    "automatically create a new pull request."
+)
+
+std_pre = (
+    "pre is an optional executable for preparing the node prior to running "
+    "the build. This step is retried on failure given that it performs "
+    "preparatory actions and not the actual function of the job. Examples of "
+    "actions that should be taken here include things that would be performed "
+    "manually such as setting configuration options."
+)
+
+std_run = (
+    "run is a required executable that executes the purpose of the job. For "
+    "example, if the project deploys some software, this file would be used "
+    "to perform the deployment and then test the success of the operation."
+)
+
+std_post = (
+    "post is an optional executable that is the last stage to run. It is used "
+    "to perform actions that make use of the data generated. Such actions "
+    "include log aggregation and preparing the node to be used as an image."
+)
+
+std_env_vars = (
+    "Environment variables\n"
+    "RE_HOOK_ARTIFACT_DIR: The directory where job artifacts (log files, "
+    "configuration used, etc) should be written.\n"
+    "RE_HOOK_RESULT_DIR: The directory where test results should be written.\n"
+    "RE_JOB_NAME: The name of the job itself. For example "
+    "PM_repo-name-series-image-scenario-action\n"
+    "RE_JOB_IMAGE: The name of the 'image' as defined in the job.\n"
+    "RE_JOB_SCENARIO: The name of the 'scenario' as defined in the job.\n"
+    "RE_JOB_ACTION: The name of the 'action' as defined in the job.\n"
+    "RE_JOB_FLAVOR: The name of the 'flavor' as defined in the job.\n"
+    "RE_JOB_TRIGGER: The name of the 'trigger' that initiated the build, "
+    "examples include 'PULL', 'PUSH' and 'TIMER'.\n"
```

Comment:
This could also be `OTHER`, though I'm not entirely sure under which conditions. Also, we have `RE_JOB_TRIGGER_DETAIL` as we discovered on Monday -- but again I'm not even sure anyone is using this or if  there's even value in setting it.